### DISCUSSION TITLE
How to get GraphiQL version?

### DISCUSSION THREAD

**Discussion by abitrolly** [graphql/graphiql#2979] [Q&A / Support]
Is there a way to see which versions of GraphiQL are deployed at https://gitlab.com/-/graphql-explorer and https://api.spacex.land/graphql/ ?

**Comment by thomasheyenbrock** [comment]
Hey @abitrolly 👋 exposing the version of GraphiQL is up to the implementor. GitLab for example uses a custom build with custom assets that doesn't allow you to deduce the specific version being used.

Others like the [official GraphiQL example](https://graphql.org/swapi-graphql) load the assets from a CDN like UNPKG. If you look at the HTML file that's loaded you'll see something like `<script src="//unpkg.com/graphiql@x.y.z/graphiql.min.js"></script>`, which tells you the version that's being used. If there is no version specifier in that link (like in the official example) that means that the latest version is being used.

**Reply by abitrolly** [reply]
It is still not clear what is the purpose of leaving this to library users? The https://github.com/graphql/graphiql/blob/main/packages/graphiql/CHANGELOG.md gives reasons whenever to fill the issue or not, and without version info embedded, I guess downstream won't be adding it too.

**Reply by thomasheyenbrock** [reply]
It's like for any other project using open source software. Take React as an example: Just by looking at the final web-app you won't know which version of `react-router` or any other dependency is being used. It's the same for the GraphiQL instance that GitLab is providing.

May I ask what exactly you mean with "whenever to fill the issue or not"?

**Reply by abitrolly** [reply]
> It's like for any other project using open source software.

I don't agree. Maybe that's true for JS world, and React in particular, but in Python (monkey)patching libraries for distribution is considered to be an anti-pattern. You can subclass library and work on your solution on top of it, but in the end if you invoke `Library.__version__` in console, you will get its version.

> May I ask what exactly you mean with "whenever to fill the issue or not"?

It is not a best experience dealing with bugs that have been already fixed in latest version, and which could be fixed with update.

**Reply by abitrolly** [reply]
To prove that JS is not different from other software in maintaining library versions, here is the (rather sad) post from [`core.js`](https://github.com/zloirock/core-js/blob/master/docs/2023-02-14-so-whats-next.md) author who shows the best practice.

![image](https://user-images.githubusercontent.com/8781107/220051039-61805c84-0198-444c-b311-9ad97fcb137e.png)

**Comment by thomasheyenbrock** [comment]
Sidenote: In previous GraphiQL versions (prior to v2) it was indeed possible to find out the GraphiQL version because we did set the main component as global, i.e. you could access `window.GraphiQL`. However, this was only ever intended for the purpose of debugging and internal testing and it was removed in v2 when we refactored that component from a class component to a function component.

**Reply by acao** [reply]
there was also a `GraphiQL.version` static getter in the `GraphiQL` component, that did not require the global, similarly to with react.

**Comment by acao** [comment]
The best place for this moving forward would be an "about" dialog with license information, current version, repo link, attributions, etc
We can inline the version with the bundler on build like React does, and provide a `GraphiQL.version` for the component as well (not global)

**Reply by acao** [reply]
https://github.com/facebook/react/blob/6b6d0617eff48860c5b4e3e79c74cbd3312cf45a/packages/react/src/React.js#L127
https://github.com/facebook/react/blob/main/packages/shared/ReactVersion.js

welp, I guess they still have a TODO to inject the version as part of the build process haha
